apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/TransactionTests.cs

```diff
@@ -133,6 +134,87 @@ public async Task DisposedTransactionDoesntLeak()
+                    {
+                        MinimumPooledSessions = 1,
+                        MaximumActiveSessions = 1,
+                        WaitOnResourcesExhausted = ResourcesExhaustedBehavior.Fail
+                    })
+            };
+            using var connection = new SpannerConnection(builder);
+            connection.Open();
+
+            using var tx1 = await connection.BeginTransactionAsync();
+            using var cmd1 = connection.CreateSelectCommand($"SELECT Int64Value FROM {_fixture.TableName} WHERE K=@k");
+            cmd1.Transaction = tx1;
+            cmd1.Parameters.Add("k", SpannerDbType.String, _key);
+            Assert.Equal(DBNull.Value, await cmd1.ExecuteScalarAsync().ConfigureAwait(false));
+            // Committing the transaction returns the session to the pool.
+            var commitTimestamp = await tx1.CommitAsync();
+            // Committing a transaction twice will return the same commit timestamp twice.
+            Assert.Equal(commitTimestamp, await tx1.CommitAsync());
+            // It is possible to get transaction properties after committing the transaction.
+            Assert.Null(tx1.ReadTimestamp);
+            Assert.NotNull(tx1.TransactionId);
+
+            // It is possible to start another transaction, as the previous session was returned to the pool.
+            using var tx2 = await connection.BeginTransactionAsync();
+            using var cmd2 = connection.CreateSelectCommand($"SELECT Int64Value FROM {_fixture.TableName} WHERE K=@k");
+            cmd2.Transaction = tx2;
+            cmd2.Parameters.Add("k", SpannerDbType.String, _key);
+            Assert.Equal(DBNull.Value, await cmd2.ExecuteScalarAsync().ConfigureAwait(false));
+            await (commit ? tx2.CommitAsync() : tx2.RollbackAsync());
+
+            // Trying to start two transactions in parallel will fail as we don't have enough sessions.
+            using var tx3 = await connection.BeginTransactionAsync();
+            var exception = await Assert.ThrowsAsync<SpannerException>(() => connection.BeginTransactionAsync());
+            Assert.Equal(ErrorCode.ResourceExhausted, exception.ErrorCode);
+
+            // Commit/rollback the transaction to return the session to the pool.
+            await (commit ? tx3.CommitAsync() : tx3.RollbackAsync());
+            // This will now work.
+            using var tx4 = await connection.BeginTransactionAsync();
+            await (commit ? tx4.CommitAsync() : tx4.RollbackAsync());
+        }
+
+        [Theory]
+        [CombinatorialData]
+        public async Task TransactionCannotBeUsedAfterCommitOrRollback(bool commit)
+        {
+            using var connection = new SpannerConnection(_fixture.ConnectionString);
+            await connection.OpenAsync();
+
+            using var tx = await connection.BeginTransactionAsync();
+            using var cmd = connection.CreateSelectCommand($"SELECT Int64Value FROM {_fixture.TableName} WHERE K=@k");
```

Comment:
Is it relevant that we're not disposing of `cmd` until the end of the method? (I love the new simplified `using` statement, but I wonder whether for this and the above test, it would be worth using the "old style" approach to make it clear when `Dispose` is called.